src/StoreApi/Schemas/AbstractAddressSchema.php

```diff
@@ -0,0 +1,181 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\StoreApi\Schemas;
+
+use Automattic\WooCommerce\Blocks\RestApi\Routes;
+
+/**
+ * AddressSchema class.
+ *
+ * Provides a generic address schema for composition in other schemas.
+ *
+ * @internal This API is used internally by Blocks--it is still in flux and may be subject to revisions.
+ * @since 4.1.0
+ */
+abstract class AbstractAddressSchema extends AbstractSchema {
+	/**
+	 * Term properties.
+	 *
+	 * @internal Note that required properties don't require values, just that they are included in the request.
+	 * @return array
+	 */
+	public function get_properties() {
+		return [
+			'first_name' => [
+				'description' => __( 'First name', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => [ 'view', 'edit' ],
+				'required'    => true,
+			],
+			'last_name'  => [
+				'description' => __( 'Last name', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => [ 'view', 'edit' ],
+				'required'    => true,
+			],
+			'company'    => [
+				'description' => __( 'Company', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => [ 'view', 'edit' ],
+				'required'    => true,
+			],
+			'address_1'  => [
+				'description' => __( 'Address', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => [ 'view', 'edit' ],
+				'required'    => true,
+			],
+			'address_2'  => [
+				'description' => __( 'Apartment, suite, etc.', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => [ 'view', 'edit' ],
+				'required'    => true,
+			],
+			'city'       => [
+				'description' => __( 'City', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => [ 'view', 'edit' ],
+				'required'    => true,
+			],
+			'state'      => [
+				'description' => __( 'State/County code, or name of the state, county, province, or district.', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => [ 'view', 'edit' ],
+				'required'    => true,
+			],
+			'postcode'   => [
+				'description' => __( 'Postal code', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => [ 'view', 'edit' ],
+				'required'    => true,
+			],
+			'country'    => [
+				'description' => __( 'Country/Region code in ISO 3166-1 alpha-2 format.', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => [ 'view', 'edit' ],
+				'required'    => true,
+			],
+		];
+	}
+
+	/**
+	 * Sanitize and format the given address object.
+	 *
+	 * @param array            $address Value being sanitized.
+	 * @param \WP_REST_Request $request The Request.
+	 * @param string           $param THe param being sanitized.
```

Comment:
Typo. :slightly_smiling_face: 

```suggestion
	 * @param string           $param The param being sanitized.
```